mylink:

  # robot_hw_test:

  #   type: robot_hw_test_ns/MyRobotInterface

  #   joints:

  #     - joint_1

  #     - joint_2

  #     - joint_3

  #     - joint_4

  #     - joint_5

  #     - joint_6

# robothw的参数


  joint_controller_test:
    type: robot_hw_test_ns/joint_controller_test
    end_effector_link: "tool0"
    robot_base_link: "base_link"
    target_frame_topic: "target_frame"
    joints:
    - joint1
    - joint2
    - joint3
    - joint4
    - joint5
    - joint6

    pd_gains:
        trans_x: {p: 10.0}
        trans_y: {p: 10.0}
        trans_z: {p: 10.0}
        rot_x: {p: 1.0}
        rot_y: {p: 1.0}
        rot_z: {p: 1.0}
  my_joint_trajectory_controller_test:
    type: franka_example_controllers/MoveToStartPosition
    end_effector_link: "tool0"
    robot_base_link: "base_link"
    target_frame_topic: "target_frame"
    joints:
    - joint1
    - joint2
    - joint3
    - joint4
    - joint5
    - joint6

    pd_gains:
        trans_x: {p: 10.0}
        trans_y: {p: 10.0}
        trans_z: {p: 10.0}
        rot_x: {p: 1.0}
        rot_y: {p: 1.0}
        rot_z: {p: 1.0}
        
  # 导纳控制器配置
  admittance_controller:
    type: robot_hw_test/AdmittanceController
    joints:
      - joint1
      - joint2
      - joint3
      - joint4
      - joint5
      - joint6
      
    # 导纳参数配置
    admittance:
      mass: 1.0
      damping: 10.0
      stiffness: 100.0

joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50